test: Do not swallow flake8 exit code

hebasto opened the pull request:
After #18210 the `flake8` exit code in `test/lint/lint-python.sh` just not used that makes the linter broken.

This PR:
- combines exit codes of `flake8` and `mypy` into the  `test/lint/lint-python.sh` exit code
- documents `mypy` as the test dependency

hebasto committed:
test: Do not swallow flake8 exit code

hebasto committed:
doc: Add mypy to test dependencies

practicalswift commented:
Oh, very nice catch! Thanks for fixing.

ACK eff770fcd299fb79f6cee93bf21f926f0f9f8251 -- patch looks correct

MarcoFalke commented on test/lint/lint-python.sh:
```diff
@@ -44,7 +44,6 @@ enabled=(
     E743 # do not define functions named "l", "O", or "I"
     E901 # SyntaxError: invalid syntax
     E902 # TokenError: EOF in multi-line string
-    F401 # module imported but unused
```
why remove this check?

hebasto replied:
https://github.com/bitcoin/bitcoin/pull/18210#issuecomment-639266292

MarcoFalke replied:
But why does travis not print this warning message?

hebasto replied:
False positive warnings due to the outdated local `flake8` version:
```
$ pip3 show flake8 | grep Version
Version: 3.5.0
```

Going to fix this pull now.

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/19172#issuecomment-639475496).

hebasto commented:
Updated eff770fcd299fb79f6cee93bf21f926f0f9f8251 -> 5d77549d8b287eb773db695b88c165ebe3be1005 ([pr19172.01](https://github.com/hebasto/bitcoin/commits/pr19172.01) -> [pr19172.02](https://github.com/hebasto/bitcoin/commits/pr19172.02), [diff](https://github.com/hebasto/bitcoin/compare/pr19172.01..pr19172.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/19172#discussion_r435887219):
> why remove this check?

MarcoFalke commented:
Approach ACK 5d77549d8b287eb773db695b88c165ebe3be1005, fine with me

practicalswift commented:
ACK 5d77549d8b287eb773db695b88c165ebe3be1005
